.github/workflows/hip-formatting.yml

```diff
@@ -0,0 +1,27 @@
+name: Clang format check
+on:
+  pull_request:
+    types: [synchronize, opened]
+    paths:
+      - 'projects/hip/**'
+      - 'projects/clr/**'
+      - 'projects/hipother/**'
+      - 'projects/hip-tests/**'
+
+jobs:
+  format:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
Prefer to pin github workflows to specific commits instead of tags (which can change).